rand-core/src/lib.rs

```diff
@@ -83,10 +83,15 @@ pub mod le;
 /// 
 /// It is recommended that implementations also implement:
 /// 
-/// - `Debug` but such that no internal details are output
-/// - `Serialize` and `Deserialize` (from Serde), possibly behind a feature gate
-/// - `Clone` if, and only if, the clone will have identical output to the original
+/// - `Debug` with a custom implementation which *does not* print any internal
+///   state (at least, `CryptoRng`s should not risk leaking state through Debug)
+/// - `Serialize` and `Deserialize` (from Serde), preferably making Serde
+///   support optional at the crate level in PRNG libs
+/// - `Clone` if, and only if, the clone will have identical output to the
+///   original (i.e. all deterministic PRNGs but not external generators)
 /// - *never* implement `Copy` (accidental copies may cause repeated values)
+/// - also *do not* implement `Default`, but instead implement `SeedableRng`
+///   thus allowing use of `rand::NewRng` (which is automatically implemented)
```

Comment:
Why not use `NewRng` for `Default`?

Reply:
PRNGs will be implemented in various crates outside of the `rand` crate, which provides `NewRng`. `Default` comes from the standard library. Therefore the locality of implementation rules make this impossible. PRNGs could implement `Default` themselves, but they'd have to depend on `rand` to do so, which I don't think they should.